include/pybind11/detail/type_caster_base.h

```diff
@@ -514,7 +515,14 @@ PYBIND11_NOINLINE std::string error_string() {
             errorString += "  " + handle(f_code->co_filename).cast<std::string>() + "("
                            + std::to_string(lineno)
                            + "): " + handle(f_code->co_name).cast<std::string>() + "\n";
-            frame = frame->f_back;
+#    if PY_VERSION_HEX >= 0x030900B1
```

Comment:
Nit: Maybe we should add a backport section details/common.h . I just wasn't sure if the recommend static inline stuff would cause any nasty ODR / LTO issues so I left it as is.